src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,37 +20,83 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the assignment identified by the index number used in the displayed assignment list.\n"
-            + "Parameters: INDEX (must be a positive integer and not greater than the size of the current "
+            + ": Deletes the assignment identified by the index number(s) used in the displayed assignment list.\n"
+            + "Parameters: INDEX [MORE INDEXES] (must be a positive integer, "
+            + "must not contain duplicates and cannot be greater than the size of the current "
             + "assignment list)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Example: " + COMMAND_WORD + " 1 2";
 
-    public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Assignment: %1$s";
+    public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Assignment(s): %1$s";
+    public static final String MESSAGE_ASSIGNMENTS_DUPLICATED_INDEX = "Duplicated indexes found.";
 
-    private final Index targetIndex;
+    private static final Comparator<Index> compareIndexes = (firstIndex, secondIndex) -> {
+        int firstIndexValue = firstIndex.getZeroBased();
+        int secondIndexValue = secondIndex.getZeroBased();
+        return secondIndexValue - firstIndexValue; // sort by descending order
+    };
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    private final List<Index> targetIndexes;
+
+    public DeleteCommand(List<Index> targetIndexes) {
+        this.targetIndexes = targetIndexes;
+    }
+
+    private static void checkForDuplicatedIndexes(List<Index> targetIndexes) throws CommandException {
+        List<Integer> zeroBasedIndexes = new ArrayList<>();
+        for (Index targetIndex : targetIndexes) {
+            int zeroBasedIndex = targetIndex.getZeroBased();
+            zeroBasedIndexes.add(zeroBasedIndex);
+        }
+
+        long distinctIndexes = zeroBasedIndexes.stream().distinct().count();
+        long lengthOfIndexesList = zeroBasedIndexes.size();
+        boolean containsDuplicates = distinctIndexes < lengthOfIndexesList;
+
+        if (containsDuplicates) {
+            throw new CommandException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand
+                            .MESSAGE_ASSIGNMENTS_DUPLICATED_INDEX));
+        }
+    }
+
+    private static void checkForInvalidIndexes(List<Index> targetIndexes, Model model) throws CommandException {
+        List<Assignment> lastShownList = model.getFilteredAssignmentList();
```

Comment:
Is there a reason these methods are static methods? They seem to only be used in the DeleteCommand class, so I don't think they have to be static.

Reply:
Amended!